### ISSUE TITLE
Reading of crds from yaml files fails with RecursionError exception.

### ISSUE THREAD

**Issue by acrom18** [sernst/kuber#1]
Hey @sernst. Wanted to say this is a great module.

When trying to read a crd from yaml files. I get a `RecursionError` exception. I have tested this with *kuber 1.7.0* on *cpython 3.7.3*. The python and yaml code used for testing are further down below. I also test it with equivalent json code.

I am unsure, if this is issue originates from kuber or some underlying module. 
```
Traceback (most recent call last):
  File "crd.py", line 5, in <module>
    .add_file('crd.yml')
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/management/__init__.py", line 279, in add_file
    version
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/management/creation.py", line 49, in from_yaml_file_multiple
    return from_yaml_multiple(f.read(), kubernetes_version)
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/management/creation.py", line 70, in from_yaml_multiple
    for d in resources_definitions.split('\n---')
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/management/creation.py", line 70, in <listcomp>
    for d in resources_definitions.split('\n---')
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/management/creation.py", line 90, in from_yaml
    return from_dict(data, kubernetes_version)
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/management/creation.py", line 171, in from_dict
    resource: Resource = resource_class()
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/latest/apiextensions_v1beta1.py", line 296, in __init__
    'spec': spec or CustomResourceDefinitionSpec(),
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/latest/apiextensions_v1beta1.py", line 887, in __init__
    'validation': validation or CustomResourceValidation(),
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/latest/apiextensions_v1beta1.py", line 1644, in __init__
    'openAPIV3Schema': open_apiv3_schema or JSONSchemaProps(),
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/latest/apiextensions_v1beta1.py", line 1843, in __init__
    'not': not_ or JSONSchemaProps(),
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/latest/apiextensions_v1beta1.py", line 1843, in __init__
    'not': not_ or JSONSchemaProps(),
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/latest/apiextensions_v1beta1.py", line 1843, in __init__
    'not': not_ or JSONSchemaProps(),
  [Previous line repeated 486 more times]
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/latest/apiextensions_v1beta1.py", line 1818, in __init__
    'additionalItems': additional_items or JSONSchemaPropsOrBool(),
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/latest/apiextensions_v1beta1.py", line 2478, in __init__
    kind='JSONSchemaPropsOrBool'
  File "/home/michael/.local/share/virtualenvs/kuber-zAnbfxPg/lib/python3.7/site-packages/kuber/definitions.py", line 44, in __init__
    self._kuber_uid = str(uuid.uuid4())
  File "/usr/lib64/python3.7/uuid.py", line 759, in uuid4
    return UUID(bytes=os.urandom(16), version=4)
  File "/usr/lib64/python3.7/uuid.py", line 153, in __init__
    if [hex, bytes, bytes_le, fields, int].count(None) != 4:
RecursionError: maximum recursion depth exceeded in comparison
```
CRD yaml file used for testing. Copied from [kubenetes.io](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/).
```
apiVersion: apiextensions/v1beta1
kind: CustomResourceDefinition
metadata:
  name: crontabs.stable.example.com
spec:
  group: stable.example.com
  versions:
    - name: v1
      served: true
      storage: true
  scope: Namespaced
  names:
    plural: crontabs
    singular: crontab
    kind: CronTab
    shortNames:
    - ct
```
The python code used:
```
resource_bundle = (
    kuber.create_bundle(kubernetes_version='latest')
    .add_file('crd.yml')
)
```

**Comment by sernst** [comment]
Hi @acrom18. Glad to hear that you like kuber and thanks for posting the issue. I haven't had much opportunity yet to exercise the apiextensions package and you definitely found a case where the Kubernetes spec has a recursive object definition that I wasn't handling correctly.

I've added a fix and released kuber version 1.7.1 that should take care of the issue. And I've added the scenario you supplied to the test suite and it passes on *cpython 3.7.3*. GIve that a try and see if works for you now and let me know if not and I'll continue to dig in.
